enhancements/baremetal/baremetal-keepalived-unicast.md

```diff
@@ -0,0 +1,184 @@
+
+The bootstrap node owns the `api-vip' during bootstrap phase because it's configured with [higher](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/manifests/baremetal/keepalived.conf.tmpl#L13) VRRP priority than the control plane nodes [priority](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml#L26).
+
+##### Control plane nodes
+
+The [Keepalived pod](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/common/baremetal/files/baremetal-keepalived.yaml) in control plane nodes already includes Keepalived and keepalived-monitor containers. 
+To support unicast mode, the keepalived-monitor should be updated to include the [bootstrap node IP address](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/monitor/dynkeepalived.go#L30) and [IP addresses of control plane nodes](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/config/node.go#L336-#L380).
+
+The bootstrap IP address is being fetched repeatedly by [GET api-vip:ip-discovery-port](https://github.com/openshift/baremetal-runtimecfg/blob/e4c115bde9c8ec32593d098e146647846cffa814/pkg/config/node.go#L185-#L200), If no reply was received from the bootstrap node Keepalived will be configured only with the control plane nodes as peers.
+
+To avoid a circular dependency (Keepalived--> api-vip--> Keepalived) the control plane nodes details retrieved from localhost:kube-apiserver and not api-vip:kube-apiserver.
+
+#### ingress-vip implementation
+
+Add unicast support for the `ingress-vip` is simpler than the `api-vip`.
+
+The following steps should be done to enable unicast for `ingress-vip`:
+1. Update the Keepalived conf template files for both [control plane nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/master/00-master/baremetal/files/baremetal-keepalived-keepalived.yaml#L49-#L73) and [compute nodes](https://github.com/openshift/machine-config-operator/blob/4c04978faf8ce36af046e18c7722d03f6e60133e/templates/worker/00-worker/baremetal/files/baremetal-keepalived-keepalived.yaml) to include the unicast details.
+2. The keepalived-monitor container should be updated to [retrieve](https://github.com/openshift/baremetal-runtimecfg/blob/8242fd84d6c6d267d2d5551abf8e4f778b4f0615/pkg/config/node.go#L219-#L244) the nodes (both compute and control) IP addresses from the kube-apiserver and render them to Keepalived config.
+
+### Risks and Mitigations
+
+It is important to not destabilize our existing installation method, that uses
+multicast successfully. Full parity is expected before landing.
+
+## Design Details
+
+### Test Plan
```

Comment:
A risk here is that we don't have upgrade coverage for bare metal IPI in CI.  We may want to make that a prerequisite to landing this work.

Reply:
can you add a note on this?

Reply:
I already added a note in the risks section regarding the upgrade, see [1].

Please let me know if you think the note should be updated.

[1] https://github.com/openshift/enhancements/blob/fb8a5aa76c8db40f53d2b363a6d9fd1ff2556c40/enhancements/baremetal/baremetal-keepalived-unicast.md#risks-and-mitigations